File: core-commons/src/main/java/net/blueshell/common/dto/event/EventDTO.java
package net.blueshell.common.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.security.Timestamp;
import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
public class EventDTO extends BaseDTO {
    private String id;
    private String title;
    private String description;
    private String location;
    private Timestamp startTime;
    private Timestamp endTime;
    private double memberPrice;
    private double publicPrice;
    private String googleId;
    private boolean visible;
    private boolean membersOnly;
    private boolean signUp;
    private List<SignUpFormDTO> signUpForm;
    private int lastEditor;
    private int committee;
    private String banner; // To validate if String
    private List<String> feedbacks;
}

File: api/src/main/kotlin/net/blueshell/api/domain/event/web/dto/EventDTO.kt
package net.blueshell.api.domain.event.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import net.blueshell.api.shared.dto.AuditedAutoIdDTO
import net.blueshell.api.survey.web.dto.SurveyDTO
import java.time.Instant

@Schema(name = "Event")
data class EventDTO(
    @field:NotNull
    var committeeId: Long? = null,

    @field:NotBlank(message = "Event title cannot be empty.")
    @field:Size(max = 255, message = "Event title cannot exceed 255 characters.")
    var title: String? = null,

    @field:NotBlank(message = "Event description cannot be empty.")
    @field:Size(max = 4095, message = "Event description cannot exceed 4095 characters.")
    var description: String? = null,

    var location: String? = null,

    @field:NotNull
    var startTime: Instant? = null,

    @field:NotNull
    var endTime: Instant? = null,

    var memberPrice: Double? = null,
    var publicPrice: Double? = null,

    @field:NotNull
    var approved: Boolean? = null,

    @field:NotNull
    var membersOnly: Boolean? = null,

    @field:NotNull
    var signUp: Boolean? = null,

    var banner: EventBannerDTO? = null,

    @field:NotNull
    var signUpCount: Long? = null,

    @field:Valid
    var signUpForm: SurveyDTO? = null
) : AuditedAutoIdDTO()
